### ISSUE TITLE
dynamic routes with loadChildren

### ISSUE THREAD

**Issue by mezzie** [angular/angular-cli#4365]
[testLazyModule.zip](https://github.com/angular/angular-cli/files/748489/testLazyModule.zip)

> Please provide us with the following information:
> ---------------------------------------------------------------

### OS?
> Windows 7, 8 or 10. Linux (which distribution). Mac OSX (Yosemite? El Capitan?)
Ubuntu 16.04

### Versions.
> Please run `ng --version`. If there's nothing outputted, please run in a Terminal: `node --version` and paste the result here:
angular-cli: 1.0.0-beta.26
node: 7.4.0
os: linux x64
@angular/common: 2.4.5
@angular/compiler: 2.4.5
@angular/core: 2.4.5
@angular/forms: 2.4.5
@angular/http: 2.4.5
@angular/platform-browser: 2.4.5
@angular/platform-browser-dynamic: 2.4.5
@angular/router: 3.4.5
@angular/compiler-cli: 2.4.5

### Repro steps.
> Was this an app that wasn't created using the CLI? What change did you do on your code? etc.
This was created using ng new

### The log given by the failure.
> Normally this include a stack trace and some more information.
EXCEPTION: Uncaught (in promise): Error: Cannot find module 'app/modules/lazy/lazy.module'.

### Mention any other details that might be useful.
When doing loadChildren from the route, it works fine but dynamically modifying the route with loadChildren isn't working

this works if I add the lazy path app.routing.ts
`
    path: 'app',

    component: HomeComponent,

    children: [

      { path: '', component: HomeOutletComponent},

      { path: 'home', component: HomeOutletComponent }

      ,{ path: "lazy", loadChildren: "app/modules/lazy/lazy.module#LazyModule"}

    ]

`
This doesn't work
`let routerConfig = this.router.config;

    if (!this.loaded) {

      let r: Route = { path: "lazy", loadChildren: "app/modules/lazy/lazy.module#LazyModule"};

      routerConfig[1].children.push(r);

      this.router.resetConfig(routerConfig);

      this.loaded = true;}`
> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by filipesilva** [comment]
Dupe of https://github.com/angular/angular-cli/issues/4234

**Comment by ersagun** [comment]
Hi thank you for your help, I solved the problem : 

I think the problem was cming from this line : **path: '', loadChildren: () => ClientComponent**

When I declared it like : **path: '', loadChildren: "./client/client.module#ClientModule"**

I parsed the file normally.  

Thank you for your helps. I recommand people who encounter this problem to execute "npm run build" systematically before trying compodoc command. Npm start launch the app but doesn't find some syntactical problems.
